### PR TITLE
A number of tweaks

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/src/common/reporters/class_reporter.dart [hunk 1]
```diff
@@ -38,33 +43,28 @@ class ClassReporter {
   }
 
   void reportClass() {
-    if (diff.containsKey('annotations'))
-      _reportList('annotations', formatter: formattedAnnotation);
+    if (diff.containsKey('annotations')) _reportList('annotations',
```
Comment (srawlins) [inline lib/src/common/reporters/class_reporter.dart]: Does this come from the formatter? This seems so much less readable to me.

File: lib/src/common/reporters/classes_reporter.dart [hunk 2]
```diff
@@ -67,18 +70,17 @@ class ClassesReporter {
         // If a class's name changed, then "name" won't be in the metadata,
         // so we'll grab it from the changed map.
         // TODO: make this better...
-        if (name == null && klass.changed.containsKey('name'))
-          name = klass.changed['name'][0];
-        if (qualifiedName == null && klass.changed.containsKey('qualifiedName'))
-          qualifiedName = klass.changed['qualifiedName'][0];
+        if (name == null && klass.changed.containsKey('name')) name =
```
Comment (srawlins) [inline lib/src/common/reporters/classes_reporter.dart]: Same here. This seems less readable.

File: lib/src/common/reporters/class_reporter.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/common/reporters/class_reporter.dart]: The general model: one line or use `{}` – which I *strongly* agree with.

File: lib/src/common/reporters/method_attributes_reporter.dart [hunk 3]
```diff
@@ -131,28 +129,26 @@ class MethodAttributesReporter {
   }
 
   String propertyListItem(String attributeName, Map property) {
-    if (attributeName == 'annotations')
-      return '* ${formattedAnnotation(property)}';
+    if (attributeName ==
+        'annotations') return '* ${formattedAnnotation(property)}';
```
Comment (srawlins) [inline lib/src/common/reporters/method_attributes_reporter.dart]: Same here, this seems less readable.

File: lib/src/common/reporters/classes_reporter.dart (hunk 2 above)
Comment (kevmoo) [inline lib/src/common/reporters/classes_reporter.dart]: ditto

File: lib/src/common/reporters/method_attributes_reporter.dart (hunk 3 above)
Comment (kevmoo) [inline lib/src/common/reporters/method_attributes_reporter.dart]: That feels like a bug...

File: lib/src/frontend/js_zip_wrapper.dart [hunk 4]
```diff
@@ -36,23 +35,35 @@ class JSZipWrapper {
...
   bool hasFile(String fileName) => zip['files'].hasProperty(fileName);
 
-  String read(String fileName) => zip.callMethod('file', [fileName]).callMethod('asText', []) as String;
+  String read(String fileName) =>
+      zip.callMethod('file', [fileName]).callMethod('asText', []) as String;
 }
 
-void getBinaryContent(uri, callback) =>
-    context['JSZipUtils'].callMethod('getBinaryContent', [uri, callback]);
+Future<ByteBuffer> getBinaryContent(uri) {
```
Comment (srawlins) [inline lib/src/frontend/js_zip_wrapper.dart]: +1 thanks much for asyncing this up! Very nice

File: web/shapeshift.dart [hunk 5]
```diff
@@ -40,61 +38,89 @@ void addToSelects(String rev) {
...
-  });
+  for (var path in versions) {
+    var versionString =
+        await HttpRequest.getString("$_storageBase/${path}VERSION");
+
+    var json = JSON.decode(versionString) as Map<String, String>;
+
+    json['channel'] = channel;
+
+    int revision = int.parse(json['revision']);
```
Comment (srawlins) [inline web/shapeshift.dart]: +1, better for sorting

File: web/shapeshift.dart [hunk 6]
```diff
@@ -40,61 +38,89 @@ void addToSelects(String rev) {
...
+  try {
+    if (goButton.disabled) {
+      throw 'Slow down!';
+    }
+    goButton.disabled = true;
+
+    diffContainer.setInnerHtml('<em>working...</em>');
+
+    int left =
+    int.parse(leftVersionSelect.selectedOptions[0].attributes['value']);
```
Comment (srawlins) [inline web/shapeshift.dart]: spacing?

File: web/shapeshift.dart [hunk 7]
```diff
@@ -40,61 +38,89 @@ void addToSelects(String rev) {
...
+    int right =
+    int.parse(rightVersionSelect.selectedOptions[0].attributes['value']);
```
Comment (srawlins) [inline web/shapeshift.dart]: spacing?

File: web/shapeshift.dart [hunk 8]
```diff
@@ -40,61 +38,89 @@ void addToSelects(String rev) {
...
+    bool includeComments = includeCommentsCheck.checked;
+
+    if (left == right) {
+      diffContainer.setInnerHtml('<em>Cannot compare the same version - $left</em>');
```
Comment (srawlins) [inline web/shapeshift.dart]: +1 thanks

File: web/shapeshift.dart [hunk 9]
```diff
@@ -40,61 +38,89 @@ void addToSelects(String rev) {
...
       '${left['revision']}/api-docs/dart-api-docs.zip';
-  String rightUri = '$storageBase/channels/${right['channel']}/release/' +
+  String rightUri = '$_storageBase/channels/${right['channel']}/release/' +
       '${right['revision']}/api-docs/dart-api-docs.zip';
 
-  getBinaryContent(rightUri, (err, rightData) {
-    //TODO: this, better
-    if (err != null)
-      throw err;
+  var rightData = await getBinaryContent(rightUri);
```
Comment (srawlins) [inline web/shapeshift.dart]: Lordy this code is so much more concise with async await.

**Comment by srawlins** [comment]
Kevin thanks so much! Can you confirm that I, uh... have to include some of those weird line re-arrangements?

This is a great cleanup though!

File: lib/src/common/reporters/class_reporter.dart (hunk 1 above)
Comment (srawlins) [inline lib/src/common/reporters/class_reporter.dart]: Hrm, ok, I'll probably find later and swap to `{ }` then.

**Pull request merged**